### ISSUE TITLE
Last thumbnail sometimes gets the wrong size

### ISSUE THREAD

**Issue by Peceha** [diffractor/diffractor#124]
Hi,
like on the picture below
the very right bottom thumbnail, marked with red V
there are two screen shots side by side (one with mouse over that thumbnail and other with mouse away)
![image](https://user-images.githubusercontent.com/67386110/187062259-f3ca9985-5c52-4d85-8bdf-e1847ef6848b.png)
It is not happening all the time (and it happens with thumbnails for video files - don't know about pictures)

Thanks

**Comment by Peceha** [comment]
In the above example the last thumbnail is most likely to wide and doesn't fit on the screen (covering the scrollbar).
And on the picture below the last thumbnail is.. to small - it should be the same size as thumbs above it.
![image](https://user-images.githubusercontent.com/67386110/187088704-1ecbb42a-cf77-4346-8bb7-49d702501def.png)

**Comment by ZacWalk** [comment]
I finally made the source code public. The method that does this layout is [here](https://github.com/diffractor/diffractor/blob/7f693727064b8b7a917a6593ae8de1342bbea4f3/src/model_items.cpp#:~:text=sizei%20df%3A%3A-,item_group%3A%3Ameasure,-\(ui%3A%3Ameasure_context%26%20mc).
I need to re-learn what that code does to fix it :)

**Comment by ZacWalk** [comment]
There is a new patch update 125.0.**1155** that should fix this issue. Also fixes some crashes.

**Comment by ZacWalk** [comment]
Yes - in this version I reduced the limit of how far an individual item can be stretched. I had other feedback that I was stretching items too much. This seems to work out better when there are more items on a line. When there are few items you get these gaps. What do you think works out better of these options:
 - Allow items to be stretched more
 - Center items when there is a gap
 - Leave as is

**Comment by Peceha** [comment]
I would vote for option (1) as that one only affects the last thumbnail

Just one more question:
when the program detects there will be a gap at the and , is there a way to crop a smaller area of the video frame with the aspect ratio of the "empty space" and then just enlarge it to fit the "empty space"?
I hope you can understand what I mean - I know ffmepg can do this since I use this method to make folder thumbnails square (without the black borders and correct aspect ratio)  in Directory Opus.

Thanks :)

**Comment by ZacWalk** [comment]
When I say stretched I actually mean it is resizing the thumb tile dimensions but cropping the image inside similar to 'object-fit: cover'. So you should not see the thumbnail distorted only cropped.

If there is a gap the logic is to expand the width of **all** thumbnails on that line. The question is how much to allow the expansion. I am not sure for the average user if it is better to have more cropping or a gap at the end.

**Comment by ZacWalk** [comment]
ok - sounds good. The cropping has been there for a long time and nobody ever complained about it. This is a case of me fixing something that was not broken :)

I don't have time to do a release this week. I will do one next week sometime to fix it (y)

**Comment by Peceha** [comment]
Hi , it's me again :)

The importance of that problem is BELOW MINIMUM :) :)

I think there must be some limit somewhere in the code which doesn't let the thumbnail zoom/crop too much.
All the thumbs are working fine now (by this I mean they fill up all the horizontal space in the row) but only when the thumbnail's panel window is wide enough.
When I start narrowing right window down, the gaps at the end of the rows start showing up (looks like some thumbnails reached kind of limits and keep their zooming/cropping size).

- Case 1 - all perfect, from very top to very bottom no gaps (more or less the windows are split in half)
![image](https://user-images.githubusercontent.com/67386110/204077451-d6f75e59-e764-436c-baab-a174d29806dc.png)

- case 2  - the right window slightly smaller then left window
notice the selected thumbnail - for some reason cannot get the size of the whole row, but the thumb below (which has different resolution aspect) gets the full row thumb
![image](https://user-images.githubusercontent.com/67386110/204077649-fd58a695-3348-4e57-b7c3-ddfdfe6db0e1.png)

- case 3 - right window getting even smaller
that thumbnail in question doesn't change its size at all while the thumb below gets adjusted to the window size (that makes me think there is some limit somewhere and cropping/zooming cannot go dipper into the picture to adjust the size)
![image](https://user-images.githubusercontent.com/67386110/204077682-6e9c031d-ddf2-44d7-a289-a78b9f31d2c6.png)

- case 4 - small left window
selected thumbnail finally got resized and looks ok but the thumb above starts making problems (means it reached the limits and the crop/zooming cannot take place anymore for that thumbnail:)
![image](https://user-images.githubusercontent.com/67386110/204078040-dc5e7a0c-8d85-49f7-8a09-8c78c69bb1b2.png)

Really not important at all :) I just like when there is one solid wall of thumbs , that's all heh.
Thanks
